pkg/clustermesh/endpointslicesync/endpointslice_test.go

```diff
@@ -156,22 +138,21 @@ func Test_meshEndpointSlice_Reconcile(t *testing.T) {
 			clusterSvc.Hostnames = map[string]string{globalSvcIP: hostname}
 		})
 
-		queue := getControllerQueue(controller)
-		require.NoError(t, waitEmptyQueue(queue))
-
-		epList, err := getEndpointSlice(&fakeClient, svcName)
-		require.NoError(t, err)
-		require.Equal(t, 1, len(epList.Items))
-		require.Equal(t, map[string]string{
-			discovery.LabelServiceName:        svcName,
-			discovery.LabelManagedBy:          utils.EndpointSliceMeshControllerName,
-			mcsapiv1alpha1.LabelSourceCluster: remoteClusterName,
-			corev1.IsHeadlessService:          "",
-		}, epList.Items[0].Labels)
-		require.Equal(t, 1, len(epList.Items[0].Endpoints))
-
-		require.NotNil(t, 1, epList.Items[0].Endpoints[0].Hostname)
-		require.Equal(t, *epList.Items[0].Endpoints[0].Hostname, hostname)
+		require.EventuallyWithT(t, func(c *assert.CollectT) {
+			epList, err := getEndpointSlice(&fakeClient, svcName)
+			assert.NoError(c, err)
+			assert.Equal(c, 1, len(epList.Items))
+			assert.Equal(c, map[string]string{
+				discovery.LabelServiceName:        svcName,
+				discovery.LabelManagedBy:          utils.EndpointSliceMeshControllerName,
+				mcsapiv1alpha1.LabelSourceCluster: remoteClusterName,
+				corev1.IsHeadlessService:          "",
+			}, epList.Items[0].Labels)
```

Comment:
This does not work as intended, because the function does not exit if the above assertion fails, hence causing an index out-of-range panic here. You should either check the return value of `assert.Equal`, or move the subsequent assertions after that `EventuallyWithT` completes.